### PR TITLE
feat: Implement clickable tag

### PR DESCRIPTION
**Pull request by ganiirsyadi**

### CODE REVIEW

**Comment by ganiirsyadi** [comment]
I updated the storybook for the normal tag and regenerated the snapshots for normal tag. Previously it renders Tag with remove Icon, that should not be the case for normal Tag.

**Comment by marthayaputra** [comment]
> I updated the storybook for the normal tag and regenerated the snapshots for normal tag. Previously it renders Tag with remove Icon, that should not be the case for normal Tag.

any reason removable tag screenshots were regenerated?

File: src/@next/Tag/Tag.tsx [hunk 1]
```diff
@@ -51,7 +65,16 @@ export const Tag = React.forwardRef<HTMLDivElement, TagProps>(function Tag(
...
   return (
-    <TagStyle ref={ref} {...props} value={value}>
+    <TagStyle
+      ref={ref}
+      {...props}
+      value={value}
+      onClick={!disabled && onClick}
+      data-clickable={!!onClick}
+      role={!!onClick ? 'button' : undefined}
+      data-disabled={!!onClick && disabled}
```
Comment (marthayaputra) [inline src/@next/Tag/Tag.tsx]: why do we need to check onClick too for disabled here?

File: src/@next/Tag/Tag.tsx [hunk 2]
```diff
@@ -14,23 +14,37 @@ export interface TagProps extends React.HTMLAttributes<HTMLDivElement> {
...
+    onClick,
+    disabled,
+    ...props
+  }: TagProps,
   ref
 ) {
+  const handleTextColor = () => {
+    if (disabled) {
+      return Neutral.B85;
+    } else if (textColor) {
```
Comment (marthayaputra) [inline src/@next/Tag/Tag.tsx]: there's no need else here if the first if is inverted, just an if will do

File: src/@next/Tag/Tag.tsx (hunk 1 above)
Comment (ganiirsyadi) [reply src/@next/Tag/Tag.tsx]: Hmm, yes, I should delete the onClick. I was thinking of someone passing `disabled` in non clickable tag. But I think that would not make any sense.

File: src/@next/Tag/Tag.tsx (hunk 2 above)
Comment (ganiirsyadi) [reply src/@next/Tag/Tag.tsx]: Fixed

File: src/@next/Tag/Tag.tsx (hunk 1 above)
Comment (ganiirsyadi) [reply src/@next/Tag/Tag.tsx]: Fixed

**Comment by ganiirsyadi** [comment]
> > I updated the storybook for the normal tag and regenerated the snapshots for normal tag. Previously it renders Tag with remove Icon, that should not be the case for normal Tag.
> 
> any reason removable tag screenshots were regenerated?

I found that the text inside tag get shifted a bit to the top, thus I translate them 1px down so it placed at the center of the tag.

Before:
image

After:
image

**Comment by marthayaputra** [comment]
> > > I updated the storybook for the normal tag and regenerated the snapshots for normal tag. Previously it renders Tag with remove Icon, that should not be the case for normal Tag.
> > 
> > 
> > any reason removable tag screenshots were regenerated?
> 
> I found that the text inside tag get shifted a bit to the top, thus I translate them 1px down so it placed at the center of the tag.
> 
> Before: image
> 
> After: image

Sorry looks like the after doesn't have any removable button?

**Comment by ganiirsyadi** [comment]
> > > > I updated the storybook for the normal tag and regenerated the snapshots for normal tag. Previously it renders Tag with remove Icon, that should not be the case for normal Tag.
> > > 
> > > 
> > > any reason removable tag screenshots were regenerated?
> > 
> > 
> > I found that the text inside tag get shifted a bit to the top, thus I translate them 1px down so it placed at the center of the tag.
> > Before: image
> > After: image
> 
> Sorry looks like the after doesn't have any removable button?

Ignore the removable button, the point is on the text itself. 
Or you can just simply check from changes and use `swipe` on snapshots difference to get what I meant


https://user-images.githubusercontent.com/52077891/220536135-009fd350-485a-40c0-8106-4829e0a236b7.mov

**Comment by marthayaputra** [comment]
> > > > > I updated the storybook for the normal tag and regenerated the snapshots for normal tag. Previously it renders Tag with remove Icon, that should not be the case for normal Tag.
> > > > 
> > > > 
> > > > any reason removable tag screenshots were regenerated?
> > > 
> > > 
> > > I found that the text inside tag get shifted a bit to the top, thus I translate them 1px down so it placed at the center of the tag.
> > > Before: image
> > > After: image
> > 
> > 
> > Sorry looks like the after doesn't have any removable button?
> 
> Ignore the removable button, the point is on the text itself. Or you can just simply check from changes and use `swipe` on snapshots difference to get what I meant

Have this changes been communicated with Maeve?

**Comment by ganiirsyadi** [comment]
> Have this changes been communicated with Maeve?

Let me have a discussion with her

**Comment by ganiirsyadi** [comment]
@marthayaputra I've discussed with Maeve, she is fine with my changes (shift the text down by 1px). It's QC approved.
https://glints.slack.com/archives/C018BD6GVS8/p1677045903420959?thread_ts=1676970121.693889&cid=C018BD6GVS8

**Review by marthayaputra — APPROVED** [review]

**Pull request merged**